### PR TITLE
add amp-install-serviceworker example

### LINKED ISSUE #119: Create amp-install-serviceworker example for AMP->PWA escalation

### PR DESCRIPTION
**Pull request by jmadler**
Resolves #119

### CODE REVIEW

File: src/20_Components/amp-install-serviceworker.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  <!--
+    The amp-install-serviceworker component requests that the runtime install a Service Worker.  This service worker could be used to cache the app shell of your Progressive Web App (PWA).
+  -->
+    <amp-install-serviceworker 
+        src="https://example.com/serviceworker.js" 
+        layout="nodisplay"></amp-install-serviceworker>
+  <!-- #### Third-Party Origin -->
+  <!--
+    If the AMP document is being served from a third-party AMP cache, such as the Google AMP Cache, you'll need to link to the URL of an HTML document to iframe in to install the service worker.
+    In this case, the HTML document could simply be an AMP HTML document itself using the amp-install-serviceworker component.
```
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: Will the iframed doc be visible? Are there any recommendations what it should include? Can we simply link to this example?

File: src/20_Components/amp-install-serviceworker.html [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+
+  <!-- #### Basic Usage -->
+  <!--
+    The amp-install-serviceworker component requests that the runtime install a Service Worker.  This service worker could be used to cache the app shell of your Progressive Web App (PWA).
+  -->
+    <amp-install-serviceworker 
+        src="https://example.com/serviceworker.js"
```
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: Can we link to an existing service worker?

File: src/20_Components/amp-install-serviceworker.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+
+  <!-- #### Basic Usage -->
+  <!--
+    The amp-install-serviceworker component requests that the runtime install a Service Worker.  This service worker could be used to cache the app shell of your Progressive Web App (PWA).
```
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: Can you add a link to https://developers.google.com/web/updates/2015/11/app-shell for app shell.

**Comment by sebastianbenz** [comment]
Nice! It would be great if we could turn this into a working example. Otherwise this doesn't really provide any benefit over the docs.

File: src/20_Components/amp-install-serviceworker.html (hunk 3 above)
Comment (jmadler) [inline src/20_Components/amp-install-serviceworker.html]: Good call!  Done.

**Comment by jmadler** [comment]
Makes sense.  I've made the changes you suggested and started a conversation ([amphtml#2983](https://github.com/ampproject/amphtml/issues/2983)) to see if it's OK to enable cross-origin SW installation for hosts other than the AMP Cache.

File: src/20_Components/amp-install-serviceworker.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  <!-- #### Basic Usage -->
+  <!--
+    The amp-install-serviceworker component requests that the runtime install a Service Worker.  This service worker could be used to cache the [app shell](https://developers.google.com/web/updates/2015/11/app-shell) of your Progressive Web App (PWA).
+  -->
+    <amp-install-serviceworker 
+        src="https://fortunate.io/web/sw.js" 
+        layout="nodisplay"></amp-install-serviceworker>
+  <!-- #### Third-Party Origin -->
+  <!--
+    If the AMP document is being served from a third-party AMP cache, such as the Google AMP Cache, you'll need to link to the URL of an HTML document to iframe in to install the service worker.
```
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: Can you link "Google AMP Cache" to the AMP Cache Sample.

**Comment by sebastianbenz** [comment]
The console shows an error:

```
Powered by AMP ⚡ HTML – Version 1461266583739
2log.js:337Uncaught Error: Did not install ServiceWorker because it does not match the current origin: https://fortunate.io/web/sw.js​​​w @ log.js:337v.error @ log.js:189T.buildCallback @ amp-install-serviceworker.js:68b.build @ custom-element.js:447V.build @ resources-impl.js:1604T.add @ resources-impl.js:357b.attachedCallback @ custom-element.js:617oc @ custom-element.js:1162H.upgradeOrRegisterElement @ custom-element.js:79nd.registerExtendedElement @ extended-element.js:28g @ runtime.js:68a.AMP.registerElement @ runtime.js:82(anonymous function) @ amp-install-serviceworker.js:120(anonymous function) @ runtime.js:154Sa @ dom.js:27Ta @ dom.js:56ie.adopt @ runtime.js:150(anonymous function) @ amp.js:57Bb.installStyles @ styles.js:72(anonymous function) @ amp.js:44(anonymous function) @ amp.js:89
log.js:337Uncaught (in promise) Error: Failed HTTP request for %s. [object Window]​​​(…)ea @ log.js:337r.createError @ log.js:202(anonymous function) @ event-helper.js:125e @ event-helper.js:55
```

File: src/20_Components/amp-install-serviceworker.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+</head>
+<body>
+
+  <!-- #### Basic Usage -->
+  <!--
+    The amp-install-serviceworker component requests that the runtime install a Service Worker.  This service worker could be used to cache the [app shell](https://developers.google.com/web/updates/2015/11/app-shell) of your Progressive Web App (PWA).
```
Comment (sebastianbenz) [inline src/20_Components/amp-install-serviceworker.html]: Should we mention that this case doesn't really make sense as pubs should always should be prepared that their AMPs are loaded via the cache?

**Pull request closed without merging**